man/systemd.network.xml

```diff
@@ -1237,6 +1237,20 @@
             <para>Allow setting custom port for the DHCP client to listen on.</para>
           </listitem>
         </varlistentry>
+
+        <varlistentry>
+          <term><varname>PromoteSecondaries=</varname></term>
+          <listitem>
+            <para>When true (the default), systemd-networkd will automatically
+            set net.ipv4.conf.$ifname.promote_secondaries to 1 if this
+            sysctl is unset so that DHCP works when the IP address
+            changes between leases. Only change this to false if you don't
+            use DHCPv4 or if your DHCPv4 never changes IPs for a given
+            machine.
```

Comment:
I am still puzzled about what this actually does. i.e. what is the precise effect of promote_secondaries being turned on? What does the kernel do differently when this is turned on?

Please do not mention the sysctl name in the man page. That#s an implementation detail they user shouldn't have to deal with.

Reply:
Thanks for your feedback (here and for the comment below). I reworded the paragraph now and hope this makes more sense now.